tools/helpers.cmake

```diff
@@ -204,30 +204,32 @@ function(GenProofsBFTarget target_name target_file pbf_path pbf_target prunes de
     )
 endfunction(GenProofsBFTarget)
 
-macro(cfg_str_add_disabled cfg_str name)
+macro(cfg_str_add_disabled cfg_str cfg_json name)
     # include in liner files can't use double slash comments
     list(APPEND ${cfg_str} "/* disabled: CONFIG_${name} */")
+    string(JSON ${cfg_json} SET ${${cfg_json}} "${name}" "null")
 endmacro()
 
-macro(cfg_str_add_define cfg_str name value comment)
+macro(cfg_str_add_define cfg_str cfg_json name value comment)
     set(cfg_define_str "#define CONFIG_${name}  ${value}")
     if(NOT "${comment}" STREQUAL "")
         # include in liner files can't use double slash comments
         string(APPEND cfg_define_str "  /* ${comment} */")
     endif()
     list(APPEND ${cfg_str} ${cfg_define_str})
+    string(JSON ${cfg_json} SET ${${cfg_json}} ${name} "\"${value}\"")
 endmacro()
 
-macro(cfg_str_add cfg_str name value)
+macro(cfg_str_add cfg_str cfg_json name value)
     if("${value}" STREQUAL "")
-        cfg_str_add_define(${cfg_str} ${name} "" "empty")
+        cfg_str_add_define(${cfg_str} ${cfg_json} ${name} "" "empty")
```

Comment:
Can't JASON use an empty string here to avoid a potential conflict if `"empty"` is ever used somewhere?

Reply:
Ah now, ignore this comment.

Reply:
Perhaps if you're unfamiliar with the cmake, this is confusing. The `"empty"` is passed to the comment parameter, not the value, so it's not semantically relevant and doesn't end up in the json.